src/sentry/api/endpoints/project_users.py

```diff
@@ -37,21 +38,82 @@ def get(self, request: Request, project) -> Response:
             project_id=project.id,
             endpoint="sentry.api.endpoints.project_users.get",
         )
-        queryset = EventUser.objects.filter(project_id=project.id)
+        # TODO(isabellaenriquez): replace from here to 50 with snubaquery and Request
+        # queryset = EventUser.objects.filter(project_id=project.id)
+        where_conditions = [Condition(Column("project_id"), Op.EQ, project.id)]
         if request.GET.get("query"):
             try:
                 field, identifier = request.GET["query"].strip().split(":", 1)
-                queryset = queryset.filter(
-                    project_id=project.id,
-                    **{EventUser.attr_from_keyword(field): identifier},
-                )
+                # queryset = queryset.filter(
+                #     project_id=project.id,
+                #     **{EventUser.attr_from_keyword(field): identifier},
+                # )
+                where_conditions.append(Condition(Column(field), Op.EQ, identifier))
             except (ValueError, KeyError):
                 return Response([])
-
+        query = Query(
+            match=Entity(EntityKey.Events.value),
+            select=[
+                Column("project_id"),
+                Column("hash"),
+                Column("ident"),
+                Column("email"),
+                Column("username"),
+                Column("name"),
+                Column("ip_address"),
+                Column("date_added"),
+            ],
+            where=where_conditions,
+        )
         return self.paginate(
             request=request,
-            queryset=queryset,
+            query=query,
+            dataset=Dataset.Events.value,
```

Comment:
wasn't sure which entity and dataset to use to properly query eventuser

Reply:
atm, getting this SnubaError for ProjectUsersTest:

`sentry.utils.snuba.SnubaError: Missing >= condition with a datetime literal on column timestamp for entity events. Example: timestamp >= toDateTime('2023-05-16 00:00')`

Reply:
@NisanthanNanthakumar yup thanks! i realized that this morning and it's in the pr now. i also learnt that the 429 errors disappeared when i passed in an org id in tenant_ids, so now i'm just trying to fix up the tests so they make sense with the new implementation